src/plugins/map/MapObject.cpp

```diff
@@ -1169,78 +1194,76 @@ void MapObjectMapLoc::UpdateMapLoc()
+			line->Color.ARGB = colorARGB;
+			line->Start.X = -m_pos.X - m_mapLocParams.lineSize;
+			line->Start.Y = -m_pos.Y - m_mapLocParams.lineSize + xWidth - 1;
 			line->Start.Z = m_pos.Z;
-			line->End.X = -m_pos.X + m_lineSize - xWidth + 1;
-			line->End.Y = -m_pos.Y + m_lineSize;
+			line->End.X = -m_pos.X + m_mapLocParams.lineSize - xWidth + 1;
+			line->End.Y = -m_pos.Y + m_mapLocParams.lineSize;
 			line->End.Z = m_pos.Z;
 			m_lines.push_back(line);
 
 			// Forwardslash lower
 			line = InitLine();
 			line->Layer = activeLayer;
-			line->Color.ARGB = m_color.ToARGB();
-			line->Start.X = -m_pos.X - m_lineSize + xWidth - 1;
-			line->Start.Y = -m_pos.Y + m_lineSize;
+			line->Color.ARGB = colorARGB;
+			line->Start.X = -m_pos.X - m_mapLocParams.lineSize + xWidth - 1;
+			line->Start.Y = -m_pos.Y + m_mapLocParams.lineSize;
 			line->Start.Z = m_pos.Z;
-			line->End.X = -m_pos.X + m_lineSize;
-			line->End.Y = -m_pos.Y - m_lineSize + xWidth - 1;
+			line->End.X = -m_pos.X + m_mapLocParams.lineSize;
+			line->End.Y = -m_pos.Y - m_mapLocParams.lineSize + xWidth - 1;
 			line->End.Z = m_pos.Z;
 			m_lines.push_back(line);
 
 			// Backslash upper
 			line = InitLine();
 			line->Layer = activeLayer;
-			line->Color.ARGB = m_color.ToARGB();
-			line->Start.X = -m_pos.X - m_lineSize + xWidth - 1;
-			line->Start.Y = -m_pos.Y - m_lineSize;
+			line->Color.ARGB = colorARGB;
+			line->Start.X = -m_pos.X - m_mapLocParams.lineSize + xWidth - 1;
+			line->Start.Y = -m_pos.Y - m_mapLocParams.lineSize;
 			line->Start.Z = m_pos.Z;
-			line->End.X = -m_pos.X + m_lineSize;
-			line->End.Y = -m_pos.Y + m_lineSize - xWidth + 1;
+			line->End.X = -m_pos.X + m_mapLocParams.lineSize;
+			line->End.Y = -m_pos.Y + m_mapLocParams.lineSize - xWidth + 1;
 			line->End.Z = m_pos.Z;
 			m_lines.push_back(line);
 
 			// Forwardslash upper
 			line = InitLine();
 			line->Layer = activeLayer;
-			line->Color.ARGB = m_color.ToARGB();
-			line->Start.X = -m_pos.X - m_lineSize;
-			line->Start.Y = -m_pos.Y + m_lineSize - xWidth + 1;
+			line->Color.ARGB = colorARGB;
+			line->Start.X = -m_pos.X - m_mapLocParams.lineSize;
+			line->Start.Y = -m_pos.Y + m_mapLocParams.lineSize - xWidth + 1;
 			line->Start.Z = m_pos.Z;
-			line->End.X = -m_pos.X + m_lineSize - xWidth + 1;
-			line->End.Y = -m_pos.Y - m_lineSize;
+			line->End.X = -m_pos.X + m_mapLocParams.lineSize - xWidth + 1;
+			line->End.Y = -m_pos.Y - m_mapLocParams.lineSize;
 			line->End.Z = m_pos.Z;
 			m_lines.push_back(line);
 		}
 	}
 
 	// Create the Radius
-	if (m_circleRadius > 0)
+	if (m_mapLocParams.circleRadius > 0)
 	{
-		m_circle.UpdateCircle(m_circleColor, m_circleRadius, m_pos.X, m_pos.Y, m_pos.Z);
+		m_circle.UpdateCircle(m_mapLocParams.circleColor, m_mapLocParams.circleRadius, m_pos.X, m_pos.Y, m_pos.Z);
 	}
 	else
 	{
 		m_circle.Clear();
 	}
+
+	delete color;
 }
 
 void MapObjectMapLoc::SetIndex(int index)
 {
-	if (mq::test_and_set(m_index, index))
-	{
-		UpdateText();
-	}
```

Comment:
why are the change checks removed?

Reply:
I always wanted it updated.  There was a bug I was running down at one point where it was failing this check when it shouldn't have.

Reply:
that means something else changed a value and you're relying on SetIndex to update the text.

Reply:
no, unfortunately the default value is 0, and I was setting it to 0 for the first time, so the text was never updated for the first entry.